pkg/trace/stats/aggregation.go

```diff
@@ -75,17 +77,22 @@ func NewAggregationFromSpan(s *pb.Span, origin string, aggKey PayloadAggregation
 			Synthetics: synthetics,
 		},
 	}
+	if peerSvcAggregation {
+		agg.PeerService = s.Meta[tagPeerService]
+	}
+	return agg
 }
 
 // NewAggregationFromGroup gets the Aggregation key of grouped stats.
 func NewAggregationFromGroup(g pb.ClientGroupedStats) Aggregation {
 	return Aggregation{
 		BucketsAggregationKey: BucketsAggregationKey{
-			Resource:   g.Resource,
-			Service:    g.Service,
-			Name:       g.Name,
-			StatusCode: g.HTTPStatusCode,
-			Synthetics: g.Synthetics,
+			Resource:    g.Resource,
+			Service:     g.Service,
+			PeerService: g.PeerService,
```

Comment:
I figured that most of the memory cost would already be accounted for in the underlying `pb.ClientGroupedStats` that we get from the tracer. I wasn't sure of the need to gate the inclusion of peer.service, since the additional overhead is the main reason we want to gate it in the traces -> concentrator path.

Reply:
Actually I meant to respond to the similar looking code in `client_stats_aggregator.go`.

In this case we don't need to worry because the grouped stats object will have `PeerService: ""` if `peer.service` aggregation has been disabled.